src/main.cpp

```diff
@@ -3555,6 +3574,13 @@ bool static LoadBlockIndexDB()
                 pindex->nChainSproutValue = pindex->nSproutValue;
             }
         }
+        // Construct in-memory chain of branch IDs.
+        // Relies on invariant: a block that does not activate a network upgrade
+        // will always be valid under the same consensus rules as its parent.
+        // Activation blocks will have non-zero branch IDs (read from disk).
+        if (pindex->IsValid(BLOCK_VALID_CONSENSUS) && pindex->nConsensusBranchId == 0 && pindex->pprev) {
```

Comment:
This line needs to be updated to change the hard-coded zero to the correct constant.